src/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+set(sources-fpp
+  clock.F90
+  ln_extract.F90
+  complexbands.F90
+  input_output.F90
+  lib_param.F90
+  elph.F90
+  phph.F90
+  ln_precision.F90
+  mpi_globals.F90
+  contselfenergy.F90
+  inversions.F90
+  libnegf.F90
+  ln_structure.F90
+  outmatrix.F90
+  rcm_module.F90
+  distributions.F90
+  ln_allocation.F90
+  load.F90
+  sparsekit_drv.F90
+  integrations.F90
+  globals.F90
+  iterative_dns.F90
+  ln_constants.F90
+  mat_def.F90
+  population.F90
+  energy_mesh.F90
+  interactions.F90
+  elphdd.F90
+  elphdb.F90
+  elphds.F90)
+
```

Comment:
It would be really nice to have here also the 2 fortran API files (`src/api/libnegfAPICommon.f90` and `src/api/libnegf_api.F90`) directly here, so we can build the library ready for C/C++ as well, but I'm having troubles suggesting a nice solution which doesn't complicate things. In principle we really just need to pick those 2 files, but I can't just do it explicitly. cmak doesn't play nice with that and the relative paths get messed up. Any idea?

Reply:
Sure, renaming ``sources`` to ``sources-fpp-f90`` and making the change:
```
set(sources-f90
  api/libnegfAPICommon.f90
  api/libnegf_api.f90)
:
add_library(negf ${sources-fpp-f90} ${sources-f90})
```
would include those files easily. However, when I tried to compile this way, I got an error message:
```
/home/aradi/sync/laptop/dftb_devel/dftbplus/external/libnegf/origin/src/api/libnegf_api.f90:834:18:

   LIB%pNEGF%minmax = minmax
                  1
Error: ‘minmax’ at (1) is not a member of the ‘tnegf’ structure; did you mean ‘emax’?
```

Reply:
I guess, we need some additional code changes here...

Reply:
It works like a charm, but it's not commited yet?